### PR TITLE
Add Searching Functionality

### PR DESCRIPTION
**Pull request by notatiyyah**
- Add Search page & Search Results page
- Create reusable search component
- Small tweaks to the header to match the [Hackney Design System](https://design-system.hackney.gov.uk/components/header#header-with-links)
- Fix Cypress randomly failing to install
- Add a sidebar to the Search/API Catalogue pages
- Tweaks to homepage to fix styling changes
- Introduce a global mobile width switch
- Tests for all new functionality
- Refactor tests by moving header & footer tests into dedicated files
- Reinstate tests that were commented out
- Add contact us page back

*Must be merged after #23 is merged*
*View changes on the [development site](https://developer-api-development.hackney.gov.uk/)*

### CODE REVIEW

File: cypress/integration/searchForApis.spec.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    screenSizes.forEach((screenSize) => {
+
+        it(`Is accessible through the sidebar on ${screenSize} screen`, function() {
+            cy.viewport(screenSize);
+            cy.visit("/api-catalogue");
+            cy.get(".sidebarLink").contains("Search").click();
+            cy.url().should("contain", "/api-catalogue/search");
+        });
+
+        it(`Shows the search bar and titleon ${screenSize} screen`, function() {
```
Comment (humulla) [inline cypress/integration/searchForApis.spec.js:19]:
```suggestion
        it(`Shows the search bar and title on ${screenSize} screen`, function() {
```

**Review by humulla — COMMENTED** [review]
Could you potentially do a test to check that you can't search if you're not signed in? (Seeing as the search icon is on the header as well)

**Review by humulla — APPROVED** [review]

**Comment by humulla** [comment]
I just realised on the dev site you added in the PR there is no search icon in the header?

**Comment by notatiyyah** [comment]
> I just realised on the dev site you added in the PR there is no search icon in the header?

@humulla I thought that was just a temporary solution while we waited for the sidebar to be implemented? I removed it now that the sidebar has been added in.

**Comment by humulla** [comment]
> > I just realised on the dev site you added in the PR there is no search icon in the header?
> 
> @humulla I thought that was just a temporary solution while we waited for the sidebar to be implemented? I removed it now that the sidebar has been added in.

Okay that's fine for now, I guess we can add it later on if we want to

**Pull request merged**